developer/src/kmc-ldml/test/compiler-e2e.tests.ts

```diff
@@ -44,6 +45,20 @@ describe('compiler-tests', function() {
     const asXml = kmxToXml(kmx);
     writeFileSync(outputFilename, asXml, 'utf-8');
 
+    // now compare it to use with run()
+    // Let's build basic.xml
+    // It should match basic.kmx (built from basic.txt)
+    const k = new LdmlKeyboardCompiler();
+    await k.init(compilerTestCallbacks, { ...compilerTestOptions, saveDebug: true, shouldAddCompilerVersion: false });
+
+    const tempdir = mkdtempSync('basic-compiler-run');
+    const outputFilename2 = join(tempdir, 'basic-run.kmx');
+
+    await k.run(inputFilename, outputFilename);
```

Comment:
Note that this does not write files out. It returns a `Promise<LdmlKeyboardCompilerResult>` which contains an `artifacts` object with `kmx?: Uint8Array` and `kvk?: Uint8Array` properties. They would not be written to disk unless you later call `LdmlKeyboardCompiler.write` with them (and even that use the callback to actually write to disk, so you can stub that out too if you need to. The compiler has no direct file system access.

Reply:
yeah, i realizeded that. However, it doesn't match basic.kmx - it contains a .kvk file and one extra store! So, challenging to test.